lib/core/src/Cardano/Wallet.hs

```diff
@@ -1613,6 +1619,27 @@ signPayment ctx wid argGenChange mkRewardAccount pwd md cs = db & \DBLayer{..} -
     tl = ctx ^. transactionLayer @t @k
     nl = ctx ^. networkLayer @t
 
+-- | Calculate the transaction expiry slot, given a 'TimeInterpreter', and an
+-- optional TTL in seconds.
+--
+-- If no TTL is provided, a default of 2 hours is used (note: there is no
+-- particular reason why we chose that duration).
+getTxExpiry
+    :: TimeInterpreter IO
+    -- ^ Context for time to slot calculation.
+    -> Maybe NominalDiffTime
+    -- ^ Time to live (TTL) in seconds from now.
+    -> IO SlotNo
+getTxExpiry ti maybeTTL = do
+    expTime <- addUTCTime ttl <$> getCurrentTime
+    -- fixme: this will explode if the user provides a TTL past the horizon.
```

Comment:
:+1:, shall we get this done in this PR? Catching the error + handling it on the server layer should be sufficient?

Reply:
Yes, shall do.